hasura/graphql-engine
# recover from SDL parse in actions type definition editor (fix #4385)

### Description



Currently, we disable the `Create` button when the SDL is in error.
In this error state, if the user makes the editor empty, we do not recover from the error and the button remains disabled.

This PR fixes this.
### Changelog


### Affected components



### Related Issues



#4385 

### Solution and Design



Empty string is technically invalid SDL. But we want to allow it, which makes it an edge case.
So we manually set the error to null if the SDL is empty.

### Steps to test and verify


Try creating action using existing types (do not declare new types). Should work.

To reproduce the exact bug that was reported:
1. Type random SDL (the editor will go into an error state)
2. Press `Ctrl + A` followed by `Del`. The editor should recover from error state.
